presto-product-tests/conf/docker/multinode-tls/docker-compose.yml

```diff
@@ -45,5 +45,5 @@ services:
       - presto-master
 
   application-runner:
-    volumes:
-      - ../../../conf/tempto/tempto-configuration-for-docker-tls.yaml:/docker/volumes/tempto/tempto-configuration-local.yaml
+    environment:
```

Comment:
Unrelated comment:

I find the structure of those docker-compose.yml files really hard to follow. The configuration is setup via a number of `compose.sh` files and it is hard to understand what is derived from what. It does not seem like having definition externally makes us be more "composition over inheritance" as we are still defining inheritance hierarchy, yet in external file.

IMO it would be nicer to have it combined so the relation between *.yml files is within files themselves. Is that possible?
What do you think?

Reply:
I have to admin that it takes time to get used to it.

Where possible "composition" is used, see how `conf/docker/common/jdbc_db.yml` is used. 

The general problem in `docker-compose` is that you often want to almost the same configuration like other configuration (e.g. singlenode), but you just want to change couple things. That way "inheritance" is much easier to do, easier to maintain, no copy-paste, hard to follow, easy to break.

This is why I recently added that whole configuration is displayed during the execution, so you exactly know what you are running.

Reply:
> That way "inheritance" is much easier to do, easier to maintain, no copy-paste, hard to follow, easy to break.

Yeah. I am fine with multiple `.yml` files which are used for composing the final one.  But maybe for inheritance (like overriding `standard.yml` with `kerberos.yml`) we could express that inheritance in `kerberos.yml` file itself.
And then use just `kerberos.yml` (not both `standard.yml` and `kerberos.yml`) in relevant profile specific docker-compose.sh? 
Expressing inheritance within `.yml` file can be done using `extension mechanism` from what I see (https://docs.docker.com/compose/extends/#extending-services)

> This is why I recently added that whole configuration is displayed during the execution, so you exactly know what you are running

Oh - I saw PR for Tempto config. Is there for docker compose also?

Reply:
> Expressing inheritance within .yml file can be done using extension mechanism

Good idea, I will give a try but in separate PR. A lot have changed around docker since we introduced `run_on_docker` product tests, maybe it is a right to time to refresh this.

> Oh - I saw PR for Tempto config. Is there for docker compose also?

https://github.com/prestodb/presto/commit/237154e950c28d0114dfa71831d3ae3670f11f71

Reply:
> Expressing inheritance within .yml file can be done using extension mechanism from what I see (https://docs.docker.com/compose/extends/#extending-services)

They suggest that there are two ways to reuse docker-compose files. One which we already following, and second which uses `extends`. The problem with `extends` is that it is not supported in `3.x` version of docker-compose. So it does not sound as a good idea to refactor to something which is going to be removed. ; /